content/en/blog/_posts/2019-12-09-csi-volume-migration-beta.md

```diff
@@ -0,0 +1,162 @@
+
+To verify the feature flag is enabled and driver installed on a particular node you can get the CSINode object. You should see the in-tree plugin name of the migrated plugin as well as your driver in the drivers list.
+
+```shell
+kubectl get csinodes -o yaml
+```
+
+```yaml
+- apiVersion: storage.k8s.io/v1
+  kind: CSINode
+  metadata:
+    annotations:
+      storage.alpha.kubernetes.io/migrated-plugins: kubernetes.io/gce-pd
+    name: test-node
+    ...
+  spec:
+    drivers:
+      name: pd.csi.storage.gke.io
+      ...
+```
+
+After the above set up is complete you can confirm that your cluster has functioning CSI migration by deploying a stateful workload using the legacy APIs.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: test-disk
+spec:
+  storageClassName: standard
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: web-server
+spec:
+  containers:
+   - name: web-server
+     image: nginx
+     volumeMounts:
+       - mountPath: /var/lib/www/html
+         name: mypvc
+  volumes:
+   - name: mypvc
+     persistentVolumeClaim:
+       claimName: test-disk
+```
+
+Verify that the pod is RUNNING after some time
+
+```shell
+kubectl get pods web-server
+```
+
+```shell
+NAME         READY   STATUS    RESTARTS   AGE
+web-server   1/1     Running   0          39s
+```
+
+To confirm that the CSI driver is actually serving your requests it may be prudent to check the container logs of the CSI Driver after exercising the storage management operations. Note that your container logs may look different depending on the provider used.
+
+```shell
+kubectl logs {CSIdriverPodName} --container={CSIdriverContainerName}
+```
+
+```shell
+/csi.v1.Controller/ControllerPublishVolume called with request: ...
+Attaching disk ... to ...
+ControllerPublishVolume succeeded for disk ... to instance ...
+```
+
+## Current limitations
+
+Although CSI migration is now beta there is one major limitation that prevents us from turning it on by default. Turning on migration still requires a cluster administrator to install a CSI driver before storage functionality is seamlessly handed over. We are currently working with SIGsig-cloudprovider to provide a frictionless experience of bundling the required CSI Drivers with cloud distributions.
+
+## What is the timeline/status?
+
+The timeline for CSI migration is actually set by the cloud provider extraction project. It is part of the effort to remove all cloud provider code from Kubernetes. By migrating cloud storage plugins to external CSI drivers we are able to extract out all the cloud provider dependencies. Since an official deprecation was announced in TODO we plan to follow the timeline and move CSI migration to GA and completely remove all in-tree plugin code from Kubernetes by v1.21.
```

Comment:
This TODO was resolved in the most recent version of the doc. Please pull in updated text